pkg/snapstore/abs_snapstore.go

```diff
@@ -386,9 +428,10 @@ func (a *ABSSnapStore) blockUploader(wg *sync.WaitGroup, stopCh <-chan struct{},
 // Delete should delete the snapshot file from store
 func (a *ABSSnapStore) Delete(snap brtypes.Snapshot) error {
 	blobName := path.Join(snap.Prefix, snap.SnapDir, snap.SnapName)
-	blob := a.containerURL.NewBlobURL(blobName)
-	if _, err := blob.Delete(context.TODO(), azblob.DeleteSnapshotsOptionInclude, azblob.BlobAccessConditions{}); err != nil {
-		return fmt.Errorf("failed to delete blob %s with error: %v", blobName, err)
+	blobClient := a.client.NewBlockBlobClient(blobName)
+	// Delete options can be mentioned once support for immutability is added
```

Comment:
I didn't get this comment what will change in `Delete op` when immutability support is added for ABS.

Reply:
These are options that can be used to either perform a soft delete or not, etc. Was just documenting it there in case it is to be used with immutability.

Reply:
ok, got it but I don't think we will ever want to do soft delete of snapshot in `Delete operation`.
Moreover, I think `soft delete` is more of functionality of versioning which azure will used in object level immutability, not in bucket level immutability.
If you want to add a comment to remember it then may be you can rephrase this little bit with proper context.

> // Delete options can be mentioned once support for immutability is added

In `Delete api` call `DeleteOptions`  can be mentioned for soft deletion if required provided versioning is enabled on container.

Reply:
I guess we can omit this comment for now. If it does become necessary we won't miss this anyway.